pyugrid/util.py

```diff
@@ -49,3 +49,28 @@ def _signed_area_tri(points):
     x3, y3 = points[2]
 
     return(((x1-x3)*(y2-y3)-(x2-x3)*(y1-y3))/2)
+
+def asarraylike(obj):
+    """
+    tests if obj acts enough like an array to be used in pyugrid. 
+
+    If it does the object is returned as is. If not, then numpy's
+    array() will be called on it.
+
+    This should catch netCDF4 variables, etc.
+
+    Note: these won't check if the attributes required actually work right.
+
+    :param obj: The object to check if it's like an array
+
+    """
+
+    must_have = ['dtype', 'shape']
```

Comment:
> does this object support nd indexing in the same way as numpy? Maybe check for `__array__` ?

Indeed!  However, I would like to have `ndim` too, I use it a lot and I guess that it makes sense to get the same "basic array info."  I am unsure if a dask array provides all that (my guess is yes), but I know that biggus does.

Maybe it is better if we make a choice and just use one of them.